pkg/networkdevice/metadata/payload.go

```diff
@@ -72,6 +82,64 @@ type DeviceMetadata struct {
 	DeviceType     string       `json:"device_type,omitempty"`
 }
 
+// DeviceOID device scan oid data
+type DeviceOID struct {
+	DeviceID string         `json:"device_id"`
+	OID      string         `json:"oid"`
+	Type     gosnmp.Asn1BER `json:"type"`
+	// Value is the stringified version of the value; if Type is OctetString or BitString, it is base64 encoded.
+	Value string `json:"value"`
+}
+
+// RawValue returns the value as a []byte, int64, float64, or plain string, depending on the type.
+func (d *DeviceOID) RawValue() (any, error) {
+	switch d.Type {
+	case gosnmp.OctetString, gosnmp.BitString:
+		return base64.StdEncoding.DecodeString(d.Value)
```

Comment:
Can't this function return an error?

Also IIRC we're doing a similar logic already in a couple places in the Agent. Would there be a way to make that a bit more generic?

Reply:
Yeah, there's a bunch of stuff here that is redundant with code in the snmp integration, but it's all under an `internal` subpackage; I tried at first to extract the pieces I needed, but it quickly became clear that I was going to need to move quite a lot of `internal` out to make that work, and I think that will need to be a separate cleanup project.

The conditional walking to get the OIDs we care about [also exists in the agent already](https://github.com/DataDog/datadog-agent/blob/b40a56ec8775a013ec7158fe9315bd93d6a9eeff/pkg/collector/corechecks/snmp/internal/session/session.go#L174), but I can't extract it without pulling the `session`, `valuestore`, `checkconfig`, `configvalidation`, and `profile` submodules out of `internal` as well.

`DecodeString` can return an error, but if it does it becomes the error returned by `RawValue`.

I think I can make this a little more generic, though, by making a PDU type for just {oid, type, value}.